### PR TITLE
Add bus voltage sensi analysis.

### PR DESCRIPTION
**Pull request by yichen88**
Signed-off-by: yichen88 <tang.yichenyves@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: cpp/src/pypowsybl.cpp [hunk 1]
```diff
@@ -289,23 +289,38 @@ void* createSensitivityAnalysis() {
...
+                targetVoltageIdPtr.get(), targetVoltageIds.size());
+}
+
 void* runSensitivityAnalysis(void* sensitivityAnalysisContext, void* network, load_flow_parameters& parameters) {
     GraalVmGuard guard;
     return executeJava<void*>(::runSensitivityAnalysis, guard.thread(), sensitivityAnalysisContext, network, &parameters);
 }
 
-matrix* getSensitivityMatrix(void* sensitivityAnalysisResultContext, const std::string& contingencyId) {
+matrix* getSensitivityMatrixFlows(void *sensitivityAnalysisResultContext, const std::string &contingencyId) {
```
Comment (yichen88) [inline cpp/src/pypowsybl.cpp]: here

File: java/src/main/java/com/powsybl/python/SensitivityAnalysisContext.java [hunk 2]
```diff
@@ -129,7 +143,13 @@ void setFactorMatrix(List<String> branchsIds, List<String> injectionsOrTransfosI
...
             }
         }
-        return factors;
+    }
+
+    private void createBusVoltageFactorMatrix(Network network, List<SensitivityFactor> factors) {
+        if (targetVoltageEquipmentsIds == null) {
+            return;
+        }
+
```
Comment (yichen88) [inline java/src/main/java/com/powsybl/python/SensitivityAnalysisContext.java:152]: TODO??? or what?

runV1 seems have no chance to run at all.

File: cpp/src/bindings.cpp [hunk 3]
```diff
@@ -340,12 +343,18 @@ PYBIND11_MODULE(_pypowsybl, m) {
                                        { sizeof(double) * m.column_count, sizeof(double) });
             });
 
-    m.def("get_sensitivity_matrix", &pypowsybl::getSensitivityMatrix, "Get sensitivity analysis result matrix for a given contingency",
+    m.def("get_sensitivity_matrix_flows", &pypowsybl::getSensitivityMatrixFlows, "Get sensitivity analysis result matrix for a given contingency",
```
Comment (sylvlecl) [inline cpp/src/bindings.cpp]: @geofjamg : do you think we should ensure retro-compatibility by keeping this kind of methods? Or is it OK in this early stage to remove some methods ?

File: cpp/src/bindings.cpp [hunk 4]
```diff
@@ -324,11 +324,14 @@ PYBIND11_MODULE(_pypowsybl, m) {
 
     m.def("create_sensitivity_analysis", &pypowsybl::createSensitivityAnalysis, "Create a sensitivity analysis");
 
-    m.def("set_factor_matrix", &pypowsybl::setFactorMatrix, "Set a factor matrix to a sensitivity analysis",
+    m.def("set_branch_flow_factor_matrix", &pypowsybl::setBranchFlowFactorMatrix, "Add a branch_flow factor matrix to a sensitivity analysis",
```
Comment (sylvlecl) [inline cpp/src/bindings.cpp:344]: I think we should be more consistent in naming between set and get methods: 
* `set_branch_flows_factor_matrix`
* `get_branch_flows_sensitivity_matrix`
* `set_bus_voltages_factor_matrix`
* `get_bus_voltages_sensitivity_matrix`

Comment (yichen88) [reply cpp/src/bindings.cpp:344]: getReferenceFlows should be changed to getFlowsReference or not?

Comment (sylvlecl) [reply cpp/src/bindings.cpp:344]: No, "reference flow" is the correct wording

**Review by sylvlecl — APPROVED** [review]

**Pull request merged**